HomeWork/src/main/java/generics/Box.java

```diff
@@ -0,0 +1,101 @@
+package generics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Box<T extends Fruit> {
+
+    private List<Fruit> fruitBox = new ArrayList<>();
+    private T fruit;
+
+    public Box(T fruit) {
+        this.fruit = fruit;
+    }
+
+
+    public T getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(T fruit) {
+        this.fruit = fruit;
+    }
+
+    public List<Fruit> getFruitBox() {
+        return fruitBox;
+    }
+
+    public void setFruitBox(List<Fruit> fruitBox) {
+        this.fruitBox = fruitBox;
+    }
+
+    public <O extends T> void add(O obj) {
+        Objects.requireNonNull(obj);
+        this.fruitBox.add(obj);
```

Comment:
You can add only the same fruit type to this box. For example, if you have box of apples, you cannot add oranges there.